code/controllers/subsystem/garbage.dm

```diff
@@ -234,14 +237,34 @@ SUBSYSTEM_DEF(garbage)
 	queue[++queue.len] = list(gctime, refid) // not += for byond reasons
 
 //this is mainly to separate things profile wise.
-/datum/controller/subsystem/garbage/proc/HardDelete(datum/D)
+/datum/controller/subsystem/garbage/proc/HardDelete(datum/D, non_datum = FALSE, override = FALSE)
+	if(!D)
+		return
+
+	if (!override && !enable_hard_deletes)
+		return
+
 	var/time = world.timeofday
 	var/tick = TICK_USAGE
 	var/ticktime = world.time
 	++delslasttick
 	++totaldels
-	var/type = D.type
-	var/refID = "\ref[D]"
+
+	var/type
+	if (!non_datum)
+		type = D.type
+	else if (islist(D))
+		type = "/list"
+	else if (istext(D))
+		type = "string"
+	else if (isnum(D))
+		type = "number"
+	else if (isfile(D))
+		type = "file"
+	else
+		type = "unknown"
```

Comment:
This should be a proc as i've seen this pattern before. would also avoid needing to pass an argument.